gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -351,17 +395,43 @@ private static GoogleCloudStorageItemInfo getObjectMetadata(
               RetryDeterminer.SOCKET_ERRORS,
               IOException.class,
               Sleeper.DEFAULT);
+      publishMetric(
+          resourceId,
+          metricsPublisher,
+          GET_OBJECT_METADATA,
+          SUCCESS,
+          JSON,
+          null,
+          "",
+          LATENCY_MS,
+          (System.currentTimeMillis() - startTime));
     } catch (IOException e) {
+      publishMetric(
+          resourceId,
+          metricsPublisher,
+          GET_OBJECT_METADATA,
+          ERROR,
+          JSON,
+          e,
+          e.getMessage(),
+          LATENCY_MS,
+          (System.currentTimeMillis() - startTime));
```

Comment:
Not sure if `System.currentTimeMillis()` is the best option to use for metrics, we may want to use `Stopwatch.elapsed()` instead.